### ISSUE TITLE
Treat click as a normal dependency

### ISSUE THREAD

**Issue by olliethomas** [guildai/guildai#247]
my project uses click for the CLI. 

When I run my script directly - no problems. When I run my script via guild there is a conflict in click - problem

This is because for some reason guildai includes click as an external dependency in `site-packages/guild/external`. This is then added to the PYTHONPATH

--- 
when I run `python my_script.py` and print the click version I get 7.1.2
when I run the same script via `guild run`  I get click version 7.0-dev

these aren't compatible
---

What I have to do is `rm -rf click*` from `site-packages/guild/external`. This is clearly not a good solution.

Can click just be treated as a normal dependency?

**Comment by gar1t** [comment]
Yes, this should be addressed. The issue for Guild is of course exposing a Guild's otherwise stable CLI to a change upstream. But to break user apps is totally unacceptable. We'll either use the normal dependency mechanism or mangle the package name that Guild uses to avoid conflicts.

This highlights the insidious problem of dependencies at all. If Guild uses the latest Click API, which is not compatible with older versions, then what of older projects? Guild will break those when installed/upgraded. We're already seeing issues come up with scipy, numpy, scikit learn, which Guild uses as well.

**Comment by tmke8** [comment]
[Poetry recommends](https://python-poetry.org/docs/#installation) being installed in its own virtual environment and is then made available globally through setting a path. Maybe that would work? Although I don't know how this interacts with injecting variables into python modules.

**Comment by gar1t** [comment]
I removed the internal use of Click so Guild will rely on the standard Python distribution requirements. This is available to try in the latest pre-release - v0.7.1.dev6. This should resolve the conflict with your program.

Re requiring that Guild is installed in a VM, that's an approach to consider. I'd prefer to keep Guild's installation as simple and as standard as possible. This could be an alternative installation technique though for those who want it.

As far as the interaction with the Python modules, that's right - Guild does use some code for that. But that could be sequestered into a small bootstrap module/package that is made available from the Guild install location without interfering with the user libraries.
